lib/mars_rover.rb

```diff
@@ -0,0 +1,15 @@
+class MarsRover
+  attr_reader :position, :direction, :commands
+
+  def initialize(x:, y:, facing_direction:, commands: [])
+    @position = Position.new(x: x, y: y)
```

Comment:
❤️

Reply:
The challenge for you (given the constraints in the README) is how you would replace the `x:` and `y:` keyword arguments with `starting point` so clients can pass a `Position` directly. That's refactoring I invite to try.